flexget/plugins/input/ombi.py

```diff
@@ -0,0 +1,248 @@
+            return access_token
+        except (RequestException, ValueError) as e:
+            raise plugin.PluginError('Ombi username and password login failed: %s' % e)
+
+    def ombi_auth(self):
+        if self.config.get('api_key'):
+            log.debug('Authenticating via api_key')
+            api_key = self.config.get('api_key')
+            header = {'apikey': api_key}
+            return header
+        elif self.config.get('username') and self.config.get('password'):
+            log.debug('Authenticating via username: %s',self.config.get('username'))
+            access_token = self.get_access_token()
+            return {"Authorization": "Bearer %s" %access_token}
+        else:
+            raise plugin.PluginError('Error: an api_key or username and password must be configured')
+
+    def get_request_list(self):
+        auth_header = self.ombi_auth()
+
+        parsedurl = urlparse(self.config.get('base_url'))
+        url=''
+        if self.config.get('type') in ['movies']:
+            url = '%s://%s:%s%s/api/v1/Request/movie' % (parsedurl.scheme, parsedurl.netloc, self.config.get('port'), parsedurl.path)
+        elif self.config.get('type') in ['shows', 'seasons', 'episodes']:
+            url= '%s://%s:%s%s/api/v1/Request/tv' % (parsedurl.scheme, parsedurl.netloc, self.config.get('port'), parsedurl.path)
+        else:
+            raise plugin.PluginError('Error: Unknown list type %s.' % (self.config.get('type')))
+        log.debug('Connecting to Ombi to retrieve request type : %s', self.config.get('type'))
+        try:
+            return requests.get(url, headers=auth_header).json()
+        except (RequestException, ValueError) as e:
+            raise plugin.PluginError('Unable to connect to Ombi at %s. Error: %s' % (url, e))
+
+    def generate_movie_entry(self, parent_request):
+        entry = Entry()
+        log.debug('Found: %s', parent_request.get('title'))
+        entry = Entry(title=self.generate_title(parent_request),
+                      url='http://www.imdb.com/title/' + parent_request.get('imdbId') +'/',
+                      imdb_id=parent_request.get('imdbId'),
+                      tmdb_id=parent_request.get('theMovieDbId'),
+                      movie_name=parent_request.get('title'),
+                      movie_year=int(parent_request.get('releaseDate')[0:4]),
+                      ombi_request_id=parent_request.get('id'),
+                      ombi_released=parent_request.get('released'),
+                      ombi_status=parent_request.get('status'),
+                      ombi_approved=parent_request.get('approved'),
+                      ombi_available=parent_request.get('available'),
+                      ombi_denied=parent_request.get('denied'))
+        return entry
+
+    def generate_tv_entry(self, parent_request, child_request=None, season=None, episode=None):
+        entry = Entry()
```

Comment:
Unused variable, put return within blocks to avoid editor complaints for returning undefined variable